frontend/app/routes/$lang+/_public+/apply+/$id+/adult/confirmation.tsx

```diff
@@ -172,6 +175,8 @@ export default function ApplyFlowConfirm() {
   const { userInfo, spouseInfo, homeAddressInfo, mailingAddressInfo, dentalInsurance, submissionInfo, csrfToken } = useLoaderData<typeof loader>();
   const powerPlatformStatusCheckerEnabled = useFeature('power-platform-status-checker');
 
+  const [modalOpen, setModalOpen] = useState(false);
```

Comment:
Use `DialogTrigger` and `DialogClose` with `Dialog` and let it manage the open and close state. At the moment you cannot close the dialog with ESC or Click the `X` close button on the dialog.

Look how to use `DialogTrigger` and `DialogClose`.
https://ui.shadcn.com/docs/components/dialog

Reply:
Removed the state and added dialog elements

Reply:
Yes.